explainer.md

```diff
@@ -68,356 +107,405 @@ If the set of pages being badged corresponds to an [Installed Web Application](h
-        status: 'ready',
-        author: 'me',
-        url: `https://example.com/status/${statusNumber}`
-      };
-
-      if (statusInfo.status === 'ready')
-        Badge.set(badgeOptions);
-      else
-        Badge.clear(badgeOptions);
-
-    </script>
-  </body>
-</html>
-```
+* `Badge.set(6, {scope: '/users/'});`
+* `Badge.set(2, {scope: '/users/1'});`
+
+Now all pages in '/users/' show the badge "6", except for `/users/1`, which
+shows the badge "2".
+
+Now if we see `Badge.clear({scope: '/users/1'})`, the pages under `/users/1`
+will start showing the badge "6" since that badge is still in effect. If instead
+we see `Badge.clear({scope: '/users/'})`, the pages under `/users/1` will still
+show the badge "2", *even if `clear` is called from one of those pages*.
+
+## Background updates
+
+For the *handle* context badges, we would like to be able to update the badge
+with a server-side push, while there are no active documents open. This would
+allow, for example, app icon badges to show an up-to-date unread count even when
+no pages are open.
+
+In this section, we explore two APIs that could be useful for this: [Push
+API](https://www.w3.org/TR/push-api/) and [Periodic Background
+Sync](https://github.com/WICG/BackgroundSync/blob/master/explainer.md#periodic-synchronization-in-design).
+
+### The Push problem
+
+The [Push API](https://www.w3.org/TR/push-api/) allows servers to send messages
+to service workers, which can run JavaScript code even when no foreground page
+is running. Thus, a server push could trigger a `Badge.set`.
+
+However, there is a [de facto standard
+requirement](https://github.com/w3c/push-api/issues/313) that whenever a push is
+received, a notification needs to be displayed. This means it's impossible to
+subtly update a badge in the background while no pages are open, without also
+displaying a notification.
+
+This may be fine for some use cases (if you want to always show a notification
+when updating the badge, possibly with an exception of not showing a
+notification if the app is in the foreground). But we specifically designed this
+API as a *subtle* notice mechanism that does not require a more distracting
+notification in order to set a badge. Because of this de facto requirement, the
+Push API currently isn't suitable for this use case.
+
+There is also a requirement that the user grants notification permission to
+subscribe to push messages.
+
+### Periodic Background Sync
+
+[Periodic Background
+Sync](https://github.com/WICG/BackgroundSync/blob/master/explainer.md#periodic-synchronization-in-design)
+is a proposed extension to the [Background
+Sync](https://wicg.github.io/BackgroundSync/spec/) API, that allows a service
+worker to periodically poll the server, which could be used to get an updated
+status and call `Badge.set`. However, this API is unreliable: the period that it
+gets called is at the discretion of the user agent and can be subject to things
+like battery status.
```

Comment:
I don't think the battery status limitation is really a problem, I'd be fine with my badges updating less frequently when I'm low on power. 

Its more that this really isn't the use case the API was designed for (download new articles an hour before the user normally opens the app or something). From a user agent's perspective, it'd be entirely reasonable to limit updates to once per `day`/`hour`